crates/engine/src/ext.rs

```diff
@@ -417,6 +419,39 @@ impl Engine {
             "off-chain environment does not yet support `call_chain_extension`"
         );
     }
+
+    pub fn ecdsa_recover(
+        &mut self,
+        signature: &[u8; 65],
+        message_hash: &[u8; 32],
+        output: &mut [u8; 33],
+    ) -> Result {
+        use secp256k1::{
+            recover,
+            Message,
+            RecoveryId,
+            Signature,
+        };
+
+        let recovery_byte = if signature[64] > 27 {
+            signature[64] - 27
+        } else {
+            signature[64]
+        };
+        let message = Message::parse(message_hash);
+        let signature = Signature::parse_slice(&signature[0..64]).unwrap();
```

Comment:
Could you replace the `unwrap()`'s in the PR with `expect(…)`? We try to avoid `unwrap`'s when possible.

Reply:
Even better is `unwrap_or_else(|error| panic!("<error description>: {}", error))` which contains the root error message alongside a local error description provided by us.